cli/azd/pkg/oneauth/oneauth_windows.go

```diff
@@ -0,0 +1,230 @@
+	if started.CompareAndSwap(false, true) {
+		err := loadDLL()
+		if err != nil {
+			return err
+		}
+		dbg := uintptr(0)
+		if debug {
+			dbg = uintptr(1)
+		}
+		clientID := unsafe.Pointer(C.CString(clientID))
+		defer C.free(clientID)
+		appID := unsafe.Pointer(C.CString(applicationID))
+		defer C.free(appID)
+		v := unsafe.Pointer(C.CString(internal.VersionInfo().Version.String()))
+		defer C.free(v)
+		p, _, _ := startup.Call(uintptr(clientID), uintptr(appID), uintptr(v), dbg)
+		// startup returns a char* message when it fails
+		if p != 0 {
+			// reset started so the next call will try to start OneAuth again
+			started.CompareAndSwap(true, false)
+			// TODO: in principle we should free the pointer but doing so from Go can cause a crash
+			msg := C.GoString((*C.char)(unsafe.Pointer(p)))
+			return fmt.Errorf("couldn't start OneAuth: %s", msg)
+		}
+	}
+	return nil
+}
+
+func authn(authority, clientID, homeAccountID, scope string, noPrompt, debug bool) (authResult, error) {
+	res := authResult{}
+	if err := start(clientID, debug); err != nil {
+		return res, err
+	}
+	a := unsafe.Pointer(C.CString(authority))
+	defer C.free(a)
+	accountID := unsafe.Pointer(C.CString(homeAccountID))
+	defer C.free(accountID)
+	// OneAuth always appends /.default to scopes
+	scope = strings.ReplaceAll(scope, "/.default", "")
+	scp := unsafe.Pointer(C.CString(scope))
+	defer C.free(scp)
+	allowPrompt := 1
+	if noPrompt {
+		allowPrompt = 0
+	}
+	p, _, _ := authenticate.Call(uintptr(a), uintptr(scp), uintptr(accountID), uintptr(allowPrompt))
+	if p == 0 {
+		// this shouldn't happen but if it did, this vague error would be better than a panic
+		return res, fmt.Errorf("authentication failed")
+	}
+	defer freeAR.Call(p)
+
+	wrapped := (*C.WrappedAuthResult)(unsafe.Pointer(p))
+	if wrapped.errorDescription != nil {
+		return res, fmt.Errorf(C.GoString(wrapped.errorDescription))
+	}
+	if wrapped.accountID != nil {
+		res.homeAccountID = C.GoString(wrapped.accountID)
+	}
+	if wrapped.token != nil {
+		res.token = azcore.AccessToken{
+			ExpiresOn: time.Unix(int64(wrapped.expiresOn), 0),
+			Token:     C.GoString(wrapped.token),
+		}
+	}
+
+	return res, nil
+}
+
+// loadDLL loads the bridge DLL and its dependencies, writing them to disk if necessary.
+func loadDLL() error {
+	if bridge != nil {
+		return nil
+	}
+	// cacheDir is %LocalAppData%
+	cacheDir, err := os.UserCacheDir()
+	if err != nil {
+		return err
+	}
+	dir := filepath.Join(cacheDir, "azd")
+	for _, dll := range []struct {
+		name, checksum string
+		data           []byte
+	}{
+		{name: "fmt.dll", checksum: fmtChecksum, data: fmtDLL},
+		{name: "bridge.dll", checksum: bridgeChecksum, data: bridgeDLL},
+	} {
+		p := filepath.Join(dir, dll.name)
+		err = writeDynamicLib(p, dll.data, dll.checksum)
+		if err != nil {
+			return fmt.Errorf("writing %s: %w", p, err)
+		}
+		// LoadLibrary searches the path for a DLL's dependencies but not the directory containing that DLL.
```

Comment:
Could we have used `LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR` instead and then just loaded `bridge.dll`?

Reply:
I believe that would work, but not with the friendly `windows.DLL` API. We'd have to call `windows.LoadLibraryEx` to set the flag and we'd get a handle back, so then it's `windows.GetProcAddress` to find functions and I think `syscall.Syscall` to call them.

Reply:
Could we call `windows.LoadLibraryEx` directly to get the handle and then construct a `windows.DLL` from it?  Looking at the state of that object it seems like we could (it is all exported).

Reply:
Yes, that does work, I've made the change. It occurred to me while doing this that there are dependencies I haven't embedded, such as MSVC runtime DLLs, which may not be present on some systems. Static linking is a much better solution than trying to embed the entire dependency graph but needs work in OneAuth, so for now I'd call this a known issue needing a fix before releasing a stable version of this feature.

Reply:
> Static linking is a much better solution than trying to embed the entire dependency graph but needs work in OneAuth, so for now I'd call this a known issue needing a fix before releasing a stable version of this feature.

My read here is that it means we would need upstream changes in order to build these libraries and statically link in the CRT and its dependencies.  Are you able to ask them to make that change for us at some point or do you need me to track down a contact and file something?

If you have an internal issue and you want me to watch it so I can make updates here once it lands, happy to do so.

Reply:
PM Scott is collecting feature requests and is aware of mine. Nothing to watch yet; I'll keep you in the loop.